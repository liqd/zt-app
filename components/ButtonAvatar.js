import React from 'react'
import { Image,TouchableOpacity } from 'react-native'

import { styles } from './ButtonAvatar.styles'

export const AvatarCircle = (props) => {
  return (
    <Image
      accessibilityIgnoresInvertColors={true}
      style={[styles.imageCircle, props.avatarStyles]}
      source={props.imgSource}
    />
  )
}

export const ButtonAvatar = (props) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress}
      accessibilityLabel={props.labelText}
      accessibilityHint={props.hintText}
      accessibilityRole="button"
      disabled={props.disabled}
    >
      <AvatarCircle
        imgSource={props.imgSource}
      />
    </TouchableOpacity>
  )
}
