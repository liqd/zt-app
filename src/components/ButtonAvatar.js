import React from 'react'
import { Image,TouchableOpacity } from 'react-native'

import { styles } from './ButtonAvatar.styles'
import { TextSourceSans } from './TextSourceSans'

export const AvatarCircle = (props) => {
  return (
    <Image
      accessibilityIgnoresInvertColors={true}
      style={[styles.imageCircle, props.avatarStyles]}
      source={props.imgSource}
    />
  )
}

export const ButtonAvatar = (props) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={props.onPress}
      accessibilityLabel={props.labelText}
      accessibilityHint={props.hintText}
      accessibilityRole="button"
      disabled={props.disabled}
    >
      <AvatarCircle
        imgSource={props.imgSource}
        avatarStyles={props.avatarStyles}
      />
      {props.field &&
      <TextSourceSans>{props.field}</TextSourceSans>
      }
    </TouchableOpacity>
  )
}
