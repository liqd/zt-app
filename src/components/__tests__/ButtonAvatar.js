import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import { ButtonAvatar } from '../ButtonAvatar'

test('Test ButtonAvatar Snapshot', () => {
  const { toJSON } = render(<ButtonAvatar />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test ButtonAvatar onPress', () => {
  const onPressMock = jest.fn()
  const { getByRole } = render(<ButtonAvatar onPress={onPressMock} disabled={false} />)
  const button = getByRole('button')
  fireEvent.press(button)
  expect(onPressMock).toHaveBeenCalled()
})

test('Test ButtonAvatar onPress disabled', () => {
  const onPressMock = jest.fn()
  const { getByRole } = render(<ButtonAvatar onPress={onPressMock} disabled={true} />)
  const button = getByRole('button')
  fireEvent.press(button)
  expect(onPressMock).not.toHaveBeenCalled()
})
