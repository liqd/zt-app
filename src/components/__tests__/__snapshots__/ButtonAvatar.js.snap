// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonAvatar Snapshot 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Image
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "backgroundColor": "#2d40cc",
          "borderRadius": 100,
          "height": 40,
          "width": 40,
        },
        undefined,
      ]
    }
  />
</View>
`;
